manage

```diff
@@ -51,19 +53,20 @@ validate_apps_gui() { #Given a list of actions and apps, graphically notify the
   for line in $queue ;do
     action="$(echo "$line" | awk -F ';' '{print $1}')"
     app="$(echo "$line" | awk -F ';' '{print $2}')"
-
+    
     if [[ "$line" =~ ^"update;filelist:".* ]];then
       # the current line is being used to pass a list of files to update
       # skip app validation
       true
-    elif [ "$action" == update ] && [ ! -d "${DIRECTORY}/update/pi-apps/apps/${app}" ];then
+    elif ([ "$action" == update ] || [ "$action" == refresh ]) && [ ! -d "${DIRECTORY}/update/pi-apps/apps/${app}" ];then
+      
       yad --class Pi-Apps --name "Pi Apps" --text="Invalid app "\""<b>$app</b>"\"". Cannot $action it." \
         --text-align=center --center --title='Error' --window-icon="${DIRECTORY}/icons/logo.png" \
         --button=OK!"${DIRECTORY}/icons/check.png":0
       
       #remove the app from list
       queue="$(echo "$queue" | sed ${i}d)"
-    elif [ "$action" != update ] && [ ! -d "${DIRECTORY}/apps/${app}" ];then
+    elif [ "$action" != update ] &&  [ "$action" != refresh ] && [ ! -d "${DIRECTORY}/apps/${app}" ];then
```

Comment:
no, this was correct before. a refresh by definition requires there to be a locally installed app. revert this part please

Reply:
I understand that was *your* definition, but I saw fit to expand the meaning to: "copying an app from update/ to apps/ when you don't want it to be reinstalled." It seems simpler this way.

Reply:
I see, you added functionality that depends on this in the next commit.

In the future it would be better to make this part of the change along with that next commit, rather than before. or note it.
https://github.com/Botspot/pi-apps/pull/2340/commits/50e4723fd826b7453f549118c53f31c74fdf096c#diff-196c23750e800642e7c008dc0416536449f295069c8161c23283c86334f8d251R323